fieldset.go

```diff
@@ -0,0 +1,43 @@
+package ucfg
+
+type fieldSet struct {
+	fields map[string]struct{}
+	parent *fieldSet
+}
+
+func makeFieldSet(parent *fieldSet) *fieldSet {
```

Comment:
for consistency use `newFieldSet`. In `go-ucfg` we use `newX` when we return a pointer and `makeX` when we return a struct value. e.g. `newFieldSet(...) *fieldSet` vs. `makeFieldSet(...) fieldSet`.